src/main/java/com/mongodb/jdbc/MongoDriver.java

```diff
@@ -278,6 +279,18 @@ public Connection connect(String url, Properties info) throws SQLException {
                         "Couldn't connect due to a timeout. Please check your hostname and port. If necessary, set a "
                                 + "longer connection timeout in the MongoDB URI.");
             } catch (Exception e) {
+                // Unwrap the cause to detect authentication failures
+                Throwable cause = e;
+                while (cause != null) {
+                    if (cause instanceof com.mongodb.MongoSecurityException) {
```

Comment:
Is it possible this is thrown for any reason other than invalid username/password?

Reply:
In the java driver it's thrown in a few places and it can be more general than username/password like `SASL protocol error: no client response to challenge for credential`.  
It is for authentication type issues so I made the error message that we show more general to say verify credentials.  I believe it's Tableau processing the sqlstate 28000 that causes it to report the username/password even though it can be a more encompassing Invalid Authorization Specialization error.